src/main/objects.c

```diff
@@ -991,16 +991,32 @@ static SEXP inherits3(SEXP x, SEXP what, SEXP which)
     else
 	PROTECT(klass = R_data_class(x, FALSE));
 
-    if(!isString(what))
-	error(_("'what' must be a character vector"));
+    int nprot = 1;
```

Comment:
Probably slightly more natural to put `int nprot = 0` at the top of the function and then increment it by 1 after the above branch